xarray/tests/test_backends.py

```diff
@@ -2913,3 +2922,22 @@ def test_dataarray_to_netcdf_no_name_pathlib(self):
 
             with open_dataarray(tmp) as loaded_da:
                 assert_identical(original_da, loaded_da)
+
+
+def test_pickle_reconstructor():
+
+    lines = ['foo bar spam eggs']
+
+    with create_tmp_file() as tmp:
```

Comment:
This needs to add `allow_cleanup_failure=ON_WINDOWS` for the test to pass on windows

Reply:
Great, that did fix the windows failure.